src/PSR7/Validators/BodyValidator/BodyValidator.php

```diff
@@ -42,38 +49,50 @@ public function validate(OperationAddress $addr, MessageInterface $message) : vo
             return;
         }
 
-        $contentTypes = $message->getHeader(self::HEADER_CONTENT_TYPE);
-        if (! $contentTypes) {
+        // Detect ContentType of the message
+        $contentType = $this->messageContentType($message);
+        if (! $contentType) {
             throw InvalidHeaders::becauseOfMissingRequiredHeader(self::HEADER_CONTENT_TYPE, $addr);
```

Comment:
We can now throw this exception directly in call above instead of returning null and checking it one more time

Reply:
True, I just did not want to pass in the `$addr` just for the sake of throwing an exception. Silly me?

Reply:
hm, sounds reasonable